build error with musl alpine

try with master branch. get this error:

```sh
h2o/lib/common/memory.c:615:12: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'char *' [-Wint-conversion]
    return strerror_r(err, buf, len);
           ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from h2o/lib/common/redis.c:23:
In file included from h2o/include/h2o/redis.h:25:
In file included fromh2o/include/h2o/socket.h:32:
/usr/include/linux/errqueue.h:57:20: error: array has incomplete element type 'struct timespec'
        struct timespec ts[3];
                          ^
/usr/include/sys/socket.h:68:8: note: forward declaration of 'struct timespec'
struct timespec;

```

Add cmake option to allow build without hiredis or libyrmcds will be great,   when use as library hredis/yrmcds is not used at many case.

when build from aarch64 ubuntu `WITHOUT_LIBS=OFF`, there is no libh2o.a, libh2o-evloop.a target.

```sh
make -j4 libh2o.a
make: *** No rule to make target 'libh2o.a'.  Stop.
```

@calvin2021y 
```

In file included from h2o/lib/common/redis.c:23:
In file included from h2o/include/h2o/redis.h:25:
In file included fromh2o/include/h2o/socket.h:32:
/usr/include/linux/errqueue.h:57:20: error: array has incomplete element type 'struct timespec'
        struct timespec ts[3];
                          ^
/usr/include/sys/socket.h:68:8: note: forward declaration of 'struct timespec'
struct timespec;
```
This error seems to be avoided
```

diff --git a/include/h2o/socket.h b/include/h2o/socket.h
index d9fbb2ac4..73bf8f7f0 100644
--- a/include/h2o/socket.h
+++ b/include/h2o/socket.h
@@ -28,7 +28,7 @@ extern "C" {
 
 #include <stdint.h>
 #include <sys/socket.h>
-#ifdef __linux__
+#ifdef __gnu_linux__
 #include <linux/errqueue.h>
 #endif
 #include <openssl/ssl.h>
```

--- a/include/h2o/socket.h
+++ b/include/h2o/socket.h
@@ -28,6 +28,7 @@ extern "C" {
 
 #include <stdint.h>
 #include <sys/socket.h>
+#include <time.h>
 #ifdef __linux__
 #include <linux/errqueue.h>
 #endif

--- a/lib/common/memory.c
+++ b/lib/common/memory.c
@@ -603,7 +603,7 @@ void h2o_append_to_null_terminated_list(void ***list, void *element)
 
 char *h2o_strerror_r(int err, char *buf, size_t len)
 {
-#ifndef _GNU_SOURCE
+#if !(defined(_GNU_SOURCE) && defined(__gnu_linux__))
     strerror_r(err, buf, len);
     return buf;
 #else

